chunky/src/java/se/llbit/chunky/renderer/scene/SampleBuffer.java

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2021 Chunky contributors
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.scene;
+
+import se.llbit.math.ColorUtil;
+
+public class SampleBuffer {
+  public final int width;   // = rowSize/3, used for alpha
+  public final int rowSize; // = width * 3, used for samples
+
+  public final int height;   // = rowCount
+  public final int rowCount; // = height
+
+  protected final double[][] samples;
+  // protected int[][] spp;     // for non-uniform sample distribution
+  protected byte[][] alpha;
+
+  public SampleBuffer(int width, int height) {
+    this.height = height;
+    this.rowCount = height;
+
+    this.width = width;
+    this.rowSize = 3 * width;
+
+    this.samples = new double[rowCount][rowSize];
+  }
+
+  public SampleBuffer(SampleBuffer toCopy) {
+    this.height = toCopy.height;
+    this.rowCount = toCopy.height;
+
+    this.width = toCopy.width;
+    this.rowSize = 3 * toCopy.width;
+
+    this.samples = new double[rowCount][rowSize];
+    for (int row = 0; row < rowCount; row++)
+      System.arraycopy(toCopy.samples[row],0,this.samples[row],0,rowSize);
+
+    if (toCopy.alpha!=null)
+    {
+      alphaInit();
+      for (int row = 0; row < rowCount; row++)
+        System.arraycopy(toCopy.alpha[row], 0, this.alpha[row], 0, width);
+    }
+  }
+
+  public long pixelCount() {
+    return width*(long)height;
+  }
+
+  public long sampleCount() {
+    return 3L*width*height;
```

Comment:
This is not the sample count. The sample count is `spp * width * height`.

Reply:
Ah, I was going for "number of double stored in this array"